### PR TITLE
Add linter, make target, and CicleCI check

### PR DESCRIPTION
**Pull request by kontaras**
- Updated the golangci-lint configuration
- Added `make lint` target
- Fixed issues caught by the linter
- Updated CircleCI config to run linter

Linter configuration is based on linter configurations from Terraform Provider Scaffolding and NuoDB Control Plane Operator projects.

### CODE REVIEW

File: internal/provider_test/integration_test.go [hunk 1]
```diff
@@ -990,7 +997,7 @@ func TestImmutableAttributeChange(t *testing.T) {
 	// Change the DBA password again, this time with
 	// NUODB_CP_ALLOW_DESTRUCTIVE_REPLACE=true
 	t.Run("applyDbaPasswordChangeWithReplace", func(t *testing.T) {
-		os.Setenv(framework.ALLOW_DESTRUCTIVE_REPLACE_VAR, "true")
+		t.Setenv(framework.ALLOW_DESTRUCTIVE_REPLACE_VAR, "true")
 		defer os.Unsetenv(framework.ALLOW_DESTRUCTIVE_REPLACE_VAR)
```
Comment (adriansuarez) [inline internal/provider_test/integration_test.go:1001]: You can drop the `defer` on line 1001, right? It seems like `t.Setenv()` does the right thing.

Comment (adriansuarez) [reply internal/provider_test/integration_test.go:1001]: Also, should `t.Setenv()` be used on the negative checks that reset `NUODB_CP_URL_BASE`? I think this is what you were commenting on in that PR, but I didn't pick up on it.

**Review by adriansuarez — APPROVED** [review]
Looks good.

File: internal/provider_test/integration_test.go [hunk 2]
```diff
@@ -855,11 +861,7 @@ func TestNegative(t *testing.T) {
 		require.Contains(t, string(out), "Invalid resource type: badresource")
 		vars.providerCfg.Timeouts = nil
 
-		// Temporarily override environment variable NUODB_CP_URL_BASE
-		// so that URL is not specified at all
```
Comment (adriansuarez) [inline internal/provider_test/integration_test.go:859]: The comment is still relevant. It's just describing the negative check that we're doing (empty string is equivalent to unspecified in terms of how we process the config).

File: go.mod [hunk 3]
```diff
@@ -15,6 +15,7 @@ require (
 
 // Test dependencies
 require (
+	github.com/golangci/golangci-lint v1.55.2
```
Comment (sivanov-nuodb) [inline go.mod:18]: Why not using the latest one `v1.56.2`?

File: internal/framework/modifiers.go [hunk 4]
```diff
@@ -3,6 +3,7 @@
 // This software is licensed under a BSD 3-Clause License.
 // See the LICENSE file provided with this software.
 
+//nolint:forcetypeassert // Type checker cannot track type guarantees in comments
```
Comment (sivanov-nuodb) [inline internal/framework/modifiers.go:6]: Are there so many lint errors in this file so that we exclude it?

File: internal/framework/schemas.go [hunk 5]
```diff
@@ -9,13 +9,14 @@ import (
 	"fmt"
 	"reflect"
 
+	"github.com/nuodb/terraform-provider-nuodbaas/openapi"
```
Comment (sivanov-nuodb) [inline internal/framework/schemas.go:12]: It is strange that the linter doesn't complain as the local imports (i.e. with prefix `github.com/nuodb/terraform-provider-nuodbaas`) should be after the 3-rd party imports.

**Review by sivanov-nuodb — APPROVED** [review]
Looks good.
I've added some minor comments on the `goimports` linting which seems to not work.

File: internal/provider_test/integration_test.go (hunk 2 above)
Comment (kontaras) [reply internal/provider_test/integration_test.go:859]: I was a bit overzealous in cutting that comment. I cut it based on how I interpreted it when I first saw it in a PR.

File: internal/framework/modifiers.go (hunk 4 above)
Comment (kontaras) [reply internal/framework/modifiers.go:6]:
```
internal/framework/modifiers.go:134:2: type assertion must be checked (forcetypeassert)
	resp.PlanValue = genericResponse.PlanValue.(types.Bool)
	^
internal/framework/modifiers.go:143:2: type assertion must be checked (forcetypeassert)
	resp.PlanValue = genericResponse.PlanValue.(types.Int64)
	^
internal/framework/modifiers.go:152:2: type assertion must be checked (forcetypeassert)
	resp.PlanValue = genericResponse.PlanValue.(types.Float64)
	^
internal/framework/modifiers.go:161:2: type assertion must be checked (forcetypeassert)
	resp.PlanValue = genericResponse.PlanValue.(types.Number)
	^
internal/framework/modifiers.go:170:2: type assertion must be checked (forcetypeassert)
	resp.PlanValue = genericResponse.PlanValue.(types.String)
	^
internal/framework/modifiers.go:179:2: type assertion must be checked (forcetypeassert)
	resp.PlanValue = genericResponse.PlanValue.(types.List)
	^
internal/framework/modifiers.go:188:2: type assertion must be checked (forcetypeassert)
	resp.PlanValue = genericResponse.PlanValue.(types.Map)
	^
internal/framework/modifiers.go:197:2: type assertion must be checked (forcetypeassert)
	resp.PlanValue = genericResponse.PlanValue.(types.Object)
```

File: go.mod (hunk 3 above)
Comment (kontaras) [reply go.mod:18]: This change has been sitting half finished in my local git for a while. I think that must have been the latest version at the time and I did not check for newer releases.

**Pull request merged**